front/components/agent_builder/capabilities/AgentBuilderCapabilitiesBlock.tsx

```diff
@@ -243,40 +237,27 @@ export function AgentBuilderCapabilitiesBlock() {
 
   const handleActionEdit = (action: AgentBuilderAction, index: number) => {
     setEditingAction({ action, index });
-
-    switch (action.type) {
-      case "SEARCH":
-        setOpenSheet("search");
-        break;
-      case "INCLUDE_DATA":
-        setOpenSheet("include_data");
-        break;
-      case "EXTRACT_DATA":
-        setOpenSheet("extract_data");
-        break;
-    }
   };
 
   const handleCloseSheet = () => {
-    setOpenSheet(null);
     setEditingAction(null);
+    setIsKnowledgeSheetOpen(false);
   };
 
-  const handleKnowledgeAdd = (serverName: string) => {
-    setEditingAction(null);
-    if (isKnowledgeServerName(serverName)) {
-      setOpenSheet(serverName);
-    } else {
-      logger.warn({ serverName }, "Unknown knowledge server");
-    }
-  };
+  const isEditingKnowledgeAction =
+    !!editingAction && isSupportedAgentBuilderAction(editingAction.action);
+
+  const shouldShowKnowledgeSheet =
+    isEditingKnowledgeAction || isKnowledgeSheetOpen;
```

Comment:
I'm confused, why we cannot track everything just with `isKnowledgeSheetOpen`? 🤔